### PR TITLE
Rackspace pagination

### PR DESCRIPTION
**Pull request by thanatos**
A set of commits to handle Rackspace's pagination of results in their DNS API in the Rackspace driver.

Originally in pull request #190, but moved here to clean it up.

### CODE REVIEW

File: libcloud/dns/drivers/rackspace.py [hunk 1]
```diff
@@ -396,6 +408,15 @@ def _ex_connection_class_kwargs(self):
         return kwargs
 
 
+def _rackspace_result_has_more(obj):
```
Comment (Kami) [inline libcloud/dns/drivers/rackspace.py]: To be on the safe side I would also pass limit and number of items returned in the response.

Then you could also return `False` if `num_items` < `limit`.

Comment (thanatos) [inline libcloud/dns/drivers/rackspace.py]: @Kami: do you mean the `totalEntries` key? (which I think is the total number, so we'd be checking if `total_entries <= offset + limit` so something like that?) (I'm using http://docs.rackspace.com/cdns/api/v1.0/cdns-devguide/content/list_domains.html as a reference.)

Comment (Kami) [inline libcloud/dns/drivers/rackspace.py]: I was thinking of number of items, so `len(records_list['records'])` for records and `len(response['domains'])` for zones.

Comment (thanatos) [inline libcloud/dns/drivers/rackspace.py]: @Kami: Oh, I see. Added a commit to that effect.

**Comment by Kami** [comment]
@thanatos Looks good. Can you please squash the commits so I can merge patch into trunk?

**Comment by thanatos** [comment]
@Kami: That's an odd request. Why not just merge the branch into `trunk`? If you really want a squashed commit (you'll lose the history?) there's one at thanatos@rackspace-pagination-squished. (1a8b287)

**Comment by Kami** [comment]
@thanatos That's a standard workflow for our project. It keeps history cleaner.

**Comment by Kami** [comment]
Patch merged into trunk. Thanks pull request can be closed now.

Thanks.

**Pull request closed without merging**